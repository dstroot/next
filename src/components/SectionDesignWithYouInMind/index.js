import React from 'react';

// import image
import womanphone from './media/womanphone.svg';

const SectionDesignWithYouInMind = () => {
  return (
    <div className="py-6">
      <div className="container text-center">
        <h1 className="nxt_heading-large">Designed with you in mind</h1>
        <img src={womanphone} alt="Woman with the Phone" />
        <h2>Fearlessly Flexible</h2>
        <p>
          Our products adapt to your lifeâ€™s biggest moments. Change them when
          your needs do.
        </p>
        <h2>Radically Simple</h2>
        <p>
          Straight forward products and features. Know what you're getting from
          the beginning.
        </p>
        <h2>Brazenly Transparent</h2>
        <p>No hidden fees or charges.</p>
      </div>
    </div>
  );
};
export default SectionDesignWithYouInMind;
