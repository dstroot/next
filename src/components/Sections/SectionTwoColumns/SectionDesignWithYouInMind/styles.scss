.para {
  position: absolute;
  top: 30%;
}
