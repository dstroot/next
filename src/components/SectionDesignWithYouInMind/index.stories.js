import React from 'react';
import { storiesOf } from '@storybook/react';
import SectionDesignWithYouInMind from '../SectionDesignWithYouInMind';

storiesOf('NEXT|sections|SectionDesignWithYouInMind', module)
  .addParameters({
    info: {
      inline: true,
      header: false,
      text: `This supports markdown!
      
      ~~~js
      console.log("hello");
      ~~~
      `,
    },
  })
  .add('show SectionDesignWithYouInMind', () => <SectionDesignWithYouInMind />);
