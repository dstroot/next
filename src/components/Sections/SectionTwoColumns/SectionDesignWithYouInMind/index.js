import React from 'react';
import { TimelineLite } from 'gsap';
import ScrollMagic from 'scrollmagic';
import 'scrollmagic/scrollmagic/uncompressed/plugins/animation.gsap';

// styles
import './styles.scss';

// component
import SectionTwoColumns from '..';

// import image
import womanphone from './media/womanphone.svg';

class SectionDesignWithYouInMind extends React.Component {
  constructor(props) {
    super(props);
    // GSAP
    this.tl = new TimelineLite();

    this.para1 = null;
    this.para2 = null;
    this.para3 = null;

    // Scrollmagic
    this.controller = new ScrollMagic.Controller();
  }

  componentDidMount() {
    // GSAP Timeline
    this.tl
      .from(this.para1, 20, { opacity: 0, y: 100 })
      .to(this.para1, 20, { opacity: 0 })
      .from(this.para2, 20, { opacity: 0, y: 100 })
      .to(this.para2, 20, { opacity: 0 })
      .from(this.para3, 20, { opacity: 0, y: 100 });

    // Scrollmagic
    new ScrollMagic.Scene({
      triggerElement: '#scrollStarts',
      duration: '600%', // increase duration of each animated scroll
      offset: 540, // start this scene after triggerElement
    })
      .setTween(this.tl)
      .setPin('#scrollStarts')
      .addTo(this.controller);
  }

  render() {
    return (
      <div>
        {/* extra div required for scrollmagic reach-router fix */}
        <div id="scrollStarts">
          <SectionTwoColumns
            styleLeft="order-2 order-sm-1"
            styleRight="order-1 order-sm-2"
            sectionHead="Designed with you in mind"
            contentLeft={
              <img
                className="img-fluid p-5"
                src={womanphone}
                alt="Woman with the Phone"
              />
            }
            contentRight={
              <div>
                <div className="para" ref={div => (this.para1 = div)}>
                  <h2 className="nxt_heading-small mb-2">
                    Fearlessly Flexible
                  </h2>
                  <p className="mb-5">
                    Our products adapt to your lifeâ€™s biggest moments. Change
                    them when your needs do.
                  </p>
                </div>
                <div className="para" ref={div => (this.para2 = div)}>
                  <h2 className="nxt_heading-small mb-2">Radically Simple</h2>
                  <p className="mb-5">
                    Straight forward products and features. Know what you're
                    getting from the beginning.
                  </p>
                </div>
                <div className="para" ref={div => (this.para3 = div)}>
                  <h2 className="nxt_heading-small mb-2">
                    Brazenly Transparent
                  </h2>
                  <p className="mb-5">No hidden fees or charges.</p>
                </div>
              </div>
            }
          />
        </div>
      </div>
    );
  }
}

export default SectionDesignWithYouInMind;
