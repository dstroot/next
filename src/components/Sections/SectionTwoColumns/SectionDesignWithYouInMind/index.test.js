import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import SectionDesignWithYouInMind from '../SectionDesignWithYouInMind';

describe('SectionDesignWithYouInMind', () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<SectionDesignWithYouInMind />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('it should render', () => {
    let tree = create(<SectionDesignWithYouInMind />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
