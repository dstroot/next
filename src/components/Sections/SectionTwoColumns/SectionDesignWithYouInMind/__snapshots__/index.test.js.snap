// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionDesignWithYouInMind it should render 1`] = `
<div
  id="triggerDWYIM"
>
  <div
<<<<<<< HEAD
    id="pinDWYIM"
=======
    id="triggerDWYIM"
>>>>>>> f19f7423847c5dc2f93b16e1d2ef3b0c2da73f90
  >
    <div
      id="pinDWYIM"
    >
      <div
        className="py-6 undefined"
      >
        <div
          className="container"
        >
          <h1
            className="react-reveal text-center nxt_heading-xlarge mb-5"
            style={
              Object {
                "opacity": undefined,
              }
            }
          >
<<<<<<< HEAD
            <img
              alt="Woman with the Phone"
              className="img-fluid p-5 nxt_mq-tall-image"
              id="dwyim-woman"
              src="womanphone.svg"
            />
          </div>
=======
            Designed with you in mind
          </h1>
>>>>>>> f19f7423847c5dc2f93b16e1d2ef3b0c2da73f90
          <div
            className="row"
          >
            <div
              className="react-reveal col-md-5 order-2 order-sm-1"
              style={
                Object {
                  "opacity": undefined,
                }
              }
            >
              <img
                alt="Woman with the Phone"
                className="img-fluid p-5 mq-tall-image"
                id="dwyim-woman"
                src="womanphone.svg"
              />
            </div>
            <div
              className="col-md-7 order-1 order-sm-2"
            >
              <div
<<<<<<< HEAD
                className="react-reveal"
                style={
                  Object {
                    "opacity": undefined,
                  }
                }
              />
              <div>
=======
                className="mt-5"
              >
>>>>>>> f19f7423847c5dc2f93b16e1d2ef3b0c2da73f90
                <div
                  className="react-reveal"
                  style={
                    Object {
                      "opacity": undefined,
                    }
                  }
                >
                  <h1
                    className="nxt_heading-xlarge"
                  />
                  <div
                    className="mt-4 pr-5"
                  />
                </div>
                <div>
                  <div
<<<<<<< HEAD
                    id="nxt_mq-para-container"
                  >
                    <div
                      className="nxt_mq-para"
                    >
                      <h2
                        className="nxt_heading-small mb-2"
                      >
                        Fearlessly Flexible
                      </h2>
                      <p
                        className="mb-5"
                      >
                        Our products adapt to your life’s biggest moments. Change them when your needs do.
                      </p>
                    </div>
                    <div
                      className="nxt_mq-para"
=======
                    className="mt-4 pr-5"
                  >
                    <div
                      id="mq-para-container"
>>>>>>> f19f7423847c5dc2f93b16e1d2ef3b0c2da73f90
                    >
                      <div
                        className="mq-para"
                      >
<<<<<<< HEAD
                        Straight forward products and features. Know what you're getting from the beginning.
                      </p>
                    </div>
                    <div
                      className="nxt_mq-para"
                    >
                      <h2
                        className="nxt_heading-small mb-2"
=======
                        <h2
                          className="nxt_heading-small mb-2"
                        >
                          Fearlessly Flexible
                        </h2>
                        <p
                          className="mb-5"
                        >
                          Our products adapt to your life’s biggest moments. Change them when your needs do.
                        </p>
                      </div>
                      <div
                        className="mq-para"
>>>>>>> f19f7423847c5dc2f93b16e1d2ef3b0c2da73f90
                      >
                        <h2
                          className="nxt_heading-small mb-2"
                        >
                          Radically Simple
                        </h2>
                        <p
                          className="mb-5"
                        >
                          Straight forward products and features. Know what you're getting from the beginning.
                        </p>
                      </div>
                      <div
                        className="mq-para"
                      >
                        <h2
                          className="nxt_heading-small mb-2"
                        >
                          Brazenly Transparent
                        </h2>
                        <p
                          className="mb-5"
                        >
                          No hidden fees or charges.
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
